stablehlo/dialect/VhloAttrs.td

```diff
@@ -203,4 +166,55 @@ def VHLO_ConvolutionAttributes {
   );
 }
 
+//===----------------------------------------------------------------------===//
+// Forked Attributes
+//===----------------------------------------------------------------------===//
+
+def VHLO_ArrayAttr : VHLO_AttrDef<"ArrayV1"> {
+  let mnemonic = "array";
+  let parameters = (ins ArrayRefParameter<"mlir::Attribute">:$value);
```

Comment:
Tried to avoid wrapping as much as possible. Rationale, in Bytecode we call into `writer.writeType`/`writer.writeAttriute` to write any nested types (nested in the sense that IntegerAttr holds an IntegerType, or TensorType has an Attribute encoding). If we wrap, we cannot use those methods and have non-standard bytecode which may cause issues. The `writeType` method logs some information, then calls into the dialect interface to write the type. It is also super useful to rely on the nested type pretty printers for numerics.

The only time I wrap currently (I believe) is IntegerType/IntegerAttr. I have heavily considered wrapping FloatAttr as well. The less we deal with numerics in VHLO the more concise it can be. You'll notice that VhloBytecode has to do special handling for IntegerAttr type as a result of the wrapping, but I think it is worth it in those cases.